src/android/BLECentralPlugin.java

```diff
@@ -1248,7 +1250,11 @@ private void findLowEnergyDevices(CallbackContext callbackContext, UUID[] servic
         discoverCallback = callbackContext;
         final BluetoothLeScanner bluetoothLeScanner = bluetoothAdapter.getBluetoothLeScanner();
         List<ScanFilter> filters = new ArrayList<ScanFilter>();
-        if (serviceUUIDs != null && serviceUUIDs.length > 0) {
+        if (this.forceScanFilter) {
```

Comment:
I'd prefer to reverse the `if` and `else if`, so that if a user sets `forceScanFilter` to true **and** supplies service UUIDs... we put the service UUIDs in first.